src/instrument/line_coverage.c

```diff
@@ -47,7 +47,7 @@ static void instrument_graph_with_breakpoints(MVMThreadContext *tc, MVMSpeshGrap
             continue;
         }
 
-        if (line_number >= 0 && line_number != last_line_number) {
+        if (line_number != last_line_number) {
             breakpoint_ins = MVM_spesh_alloc(tc, g, sizeof(MVMSpeshIns));
```

Comment:
I think that it's wrong here.

    MVMint64 line_number = -1;

That's a signed value. What does it think that `line_number` can't be less than zero? To me it looks possible to get to the `if` statement without `line_number` being changed.

Reply:
To me it looks like control only gets here if the `if` block starting on line 25 is run (since its `else` has a `continue`), and line 26 is `line_number = bbba->line_number;` and `bbba->line_number` is an `MVMuint32` https://github.com/MoarVM/MoarVM/blob/master/src/core/bytecode.h#L5.

Reply:
Yes, I think that you're correct. Thanks for that analysis.

Looks like the signed/unsigned oversight has been present since that code was first added in commit 24af1679de

I'd not considered this directly before, but I'm thinking that having `continue` in an `else` block is something to be avoided. So (in a future commit) we change

```
        if (bbba) {
            line_number = bbba->line_number;
            filename_string_index = bbba->filename_string_heap_index;
            MVM_free(bbba);
        } else {
            bb = bb->linear_next;
            continue;
        }
```

to this:

```
        if (!bbba) {
            bb = bb->linear_next;
            continue;
        }

        line_number = bbba->line_number;
        filename_string_index = bbba->filename_string_heap_index;
        MVM_free(bbba);
```

(and more generally I'm going to do this in every language in every codebase, unless there's a better reason not to. For all control flow keywords.)